nixos/doc/manual/development/option-declarations.xml

```diff
@@ -65,4 +65,68 @@ options = {
 
 </para>
 
+<section xml:id="sec-option-declarations-eot"><title>Extended Option 
+    Types</title>
```

Comment:
nit: I would replace `Extended` by `Extensible`, and do the same for all the references below as well.
`Extended` would apply to the result, not the property of a type.